src/helpers/task-metadata.ts

```diff
@@ -0,0 +1,82 @@
+import { DateTime } from "luxon";
+import { Context } from "../types/context";
+import { ListCommentsForIssue, ListForOrg, ListIssueForRepo } from "../types/github-types";
+import ms from "ms";
+
+export async function handleCommentsAndMetadata(
+    context: Context,
+    repo: ListForOrg["data"][0],
+    issue: ListIssueForRepo
+) {
+    const { logger, octokit } = context;
+
+    const comments = (await octokit.paginate(octokit.rest.issues.listComments, {
+        owner: repo.owner.login,
+        repo: repo.name,
+        issue_number: issue.number,
+        per_page: 100,
+    })) as ListCommentsForIssue[];
+
+    const botComments = comments.filter((o) => o.user?.type === "Bot");
+    const taskDeadlineJsonRegex = /"taskDeadline": "([^"]*)"/g;
+    const taskAssigneesJsonRegex = /"taskAssignees": \[([^\]]*)\]/g;
+    const assignmentRegex = /Ubiquity - Assignment - start -/gi;
+    const botAssignmentComments = botComments.filter(
+        (o) => assignmentRegex.test(o?.body || "")
+    ).sort((a, b) =>
+        DateTime.fromISO(a.created_at).toMillis() - DateTime.fromISO(b.created_at).toMillis()
+    )
+
+    const botFollowup = /this task has been idle for a while. Please provide an update./gi;
```

Comment:
Match with metadata header instead

Reply:
I'll need to take the function from `command-start-stop` and create a new comment as it's posted by this plugin